pkg/server/server.go

```diff
@@ -0,0 +1,162 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/gorilla/mux"
+	"github.com/xenitab/azdo-proxy/pkg/auth"
+)
+
+type AzdoServer struct {
+	logger logr.Logger
+	port   string
+	proxy  *httputil.ReverseProxy
+	authz  auth.Authorization
+}
+
+func NewAzdoServer(logger logr.Logger, port string, authz auth.Authorization) *AzdoServer {
+	director := func(req *http.Request) {
+		token, err := tokenFromRequest(req)
+		if err != nil {
+			logger.Error(err, "token not present in request")
+			return
+		}
+		target, err := authz.TargetForToken(token)
+		if err != nil {
+			logger.Error(err, "could not get target for token")
+			return
+		}
+
+		targetQuery := target.RawQuery
+		req.URL.Scheme = target.Scheme
+		req.URL.Host = target.Host
+		req.URL.Path, req.URL.RawPath = joinURLPath(target, req.URL)
+		if targetQuery == "" || req.URL.RawQuery == "" {
```

Comment:
I'm a bit unsure on how the URL looks like. A comment with a example to make it more clear with the if and the else on how the string should look like.